cinder/tests/unit/test_sheepdog.py

```diff
@@ -103,17 +106,50 @@ def fake_stats(*args):
         self.assertDictMatch(expected, actual)
 
     def test_check_for_setup_error_0_5(self):
-        def fake_stats(*args):
+        def fake_stats(*command, **kwargs):
             return COLLIE_CLUSTER_INFO_0_5, ''
-        self.stubs.Set(self.driver, '_execute', fake_stats)
+        self.stubs.Set(self.driver, '_command_execute', fake_stats)
         self.driver.check_for_setup_error()
 
     def test_check_for_setup_error_0_6(self):
-        def fake_stats(*args):
+        def fake_stats(*command, **kwargs):
             return COLLIE_CLUSTER_INFO_0_6, ''
-        self.stubs.Set(self.driver, '_execute', fake_stats)
+        self.stubs.Set(self.driver, '_command_execute', fake_stats)
         self.driver.check_for_setup_error()
 
+    def test_check_for_setup_error_waiting_format(self):
+        def fake_stats(*command, **kwargs):
+            return COLLIE_CLUSTER_INFO_WAITING_FORMAT, ''
+        self.stubs.Set(self.driver, '_command_execute', fake_stats)
+        self.assertRaises(exception.VolumeBackendAPIException,
+            self.driver.check_for_setup_error)
+
+    def test_check_for_setup_error_cmd_notfound(self):
+        def fake_stats(*command, **kwargs):
+            raise exception.SheepdogCmdException(message='message',
+                    cmd='command', code=127, out='stdout', err='stderr')
+        self.stubs.Set(self.driver, '_command_execute', fake_stats)
+        self.assertRaises(exception.VolumeBackendAPIException,
+            self.driver.check_for_setup_error)
```

Comment:
with self.assertRaises(exception.VolumeBackendAPIException) as cm:
    self.driver.check_for_setup_error()

msg = _LE('Sheepdog is not installed.')
self.assertEqual(msg, cm.exception.data)